gammapy/datasets/make.py

```diff
@@ -64,3 +70,90 @@ def sigma_energy_theta(energy, theta, sigma):
+
+    Returns
+    -------
+    obs_table : `~gammapy.obs.ObservationTable`
+    	observation table
+    """
+    n_obs_start = 1
+
+    astro_table = Table()
+
+    # obs id
+    obs_id = np.arange(n_obs_start, n_obs_start + n_obs)
+    col_obs_id = Column(name='OBS_ID', data=obs_id)
+    astro_table.add_column(col_obs_id)
+
+    # on time
+    ontime = Quantity(30.*np.ones_like(col_obs_id.data), 'minute')
+    col_ontime = Column(name='ONTIME', data=ontime)
+    astro_table.add_column(col_ontime)
+
+    # livetime
+    livetime = Quantity(25.*np.ones_like(col_obs_id.data), 'minute')
+    col_livetime = Column(name='LIVETIME', data=livetime)
+    astro_table.add_column(col_livetime)
+
+    # TODO: add columns for coordinates:
+    #       pointing observation -> done
+    #       and date -> done
+    #       then convert to alt az, considering the observatory location
+    # TODO: add column(s) for offset, and take it into account in the coord. transformation!!!
+
+    # RA, Dec
+    # random points on a sphere ref: http://mathworld.wolfram.com/SpherePointPicking.html
+
+    ra = Angle(360.*np.random.random(len(col_obs_id)), 'degree')
+    col_ra = Column(name='RA', data=ra)
+    astro_table.add_column(col_ra)
+
+    dec = Angle(np.arccos(2.*np.random.random(len(col_obs_id)) - 1), 'radian').to('degree')
+    # translate angles from [0, 180) deg to [-90, 90) deg
+    dec = dec - Angle(90., 'degree')
+    col_dec = Column(name='DEC', data=dec)
+    astro_table.add_column(col_dec)
+
+    # date
+    # random points between the start of 2010 and the end of 2014
+    #TODO: should this represent the time at the beginning of the run?
+    #      this has consequences for the ra/dec -> alt/az conversion
+    datestart = Time('2010-01-01T00:00:00', format='fits', scale='utc')
+    dateend = Time('2015-01-01T00:00:00', format='fits', scale='utc')
+    date = Time((dateend.mjd - datestart.mjd)*np.random.random(len(col_obs_id)) + datestart.mjd, format='mjd', scale='utc').fits
+    col_date = Column(name='DATE', data=date)
+    astro_table.add_column(col_date)
+
+
+    # alt, az
+    # TODO: should they be in the ObservationTable? (they are derived quantities, like dead time)
+    # TODO: since I randomized RA/DEC without taking into account the observatory, I'm getting negative altitudes!!!
+    #       maybe I should randomize alt az, then transform to RA/DEC!!!
+    observatory_location = observatory_locations[observatory]
+
+    # ref: http://astropy.readthedocs.org/en/latest/coordinates/observing-example.html
+    sky_coord = SkyCoord(astro_table['RA'], astro_table['DEC'], frame='icrs')
+    alt_az_coord = sky_coord.transform_to(AltAz(obstime=astro_table['DATE'], location=observatory_location))
+    #print(alt_az_coord)
+    #col_alt = alt_az_coord...
+
+
+    # TODO: operate row by row (using Observation class) instead of by columns?
+    # TODO: general methods for filling obs tables run by run; this function should call it.
```

Comment:
Generally it should be preferred to have vectorized functions. If something can't be vectorised, then usually the right strategy is to put the Python for look as low as possible in the utility functions that are called.
But this can be tricky and sometimes it is the right decision to even have user-facing functions that are not vectorised.
Here I don't see any issue with using vectorised expressions everywhere?
We can talk about this tonight.